packages/block-editor/src/components/off-canvas-editor/block-select-button.js

```diff
@@ -60,6 +61,13 @@ function ListViewBlockSelectButton(
 		}
 	}
 
+	// This is a very annoying kind of exception.
+	// Maybe we need to have a config that allows blocks
+	// to have a different icon in the list view?
```

Comment:
Let's do some dependency inversion on this.

I think we could consider something akin to [`__experimentalLabel`](https://github.com/WordPress/gutenberg/blob/b18757cbc23ceed68cb7c8df3d5d867785bbee3c/packages/block-library/src/heading/index.js#L34).

Basically I'm suggesting a way to dynamically fetch the icon which the _block_ is responsible for defining. Perhaps by passing in a `context` we can allow the block to determine which icon to display.

```js
__experimentalIcon( attributes, { context } ) {
	const { content, level } = attributes;


	if ( context === 'list-view' && content ) {
		// return the submenu icon
	}

	return // standard icon
},
```